Examples/Image/TransferLearning/TransferLearning_distributed.py

```diff
@@ -0,0 +1,275 @@
+
+    args = vars(parser.parse_args())
+    VERBOSE = not args["quiet"]
+
+
+    if not (args.get("train_map_file", False) or 
+            args.get("evaluate_only", False)):
+        print("\n ERROR: Choose whether to transfer learning on a new model "
+              "(`-tr' option) or to evaluate the performance of the model on "
+              "some images (`-ev' option).",
+              file=sys.stderr)
+        exit(1)
+    elif (distributed.Communicator.num_workers() < 2 and 
+          distributed.Communicator.rank() == 0):
+        print("\n ERROR: Run this script with at least two workers.",
+              file=sys.stderr)
+        exit(1)
+
+    distributed_after = args.get("distributed_after", False)    
+    if not distributed_after:
+        # Read the amount the user inserted, or set it to one epoch-size.
+        # -1 stands for "distribute after one epoch size".
+        distributed_after = -1
+
+    with open(args["labels_list_file"]) as cmf:
+        class_mapping = cmf.readline()
+        if not class_mapping and distributed.Communicator.rank() == 0:
+            print("\n ERROR: empty class mapping file '{}'.".format(
+                args["labels_list_file"]), file=sys.stderr)
+            exit(1)
+
+    class_mapping = np.asarray(class_mapping.strip("[] ").split(", "))
+
+    if not args["evaluate_only"]:
+        if not args["disable_timing"] and distributed.Communicator.rank() == 0:
+            start = time.clock()
+
+        # Distributed retraining.
+        trained_model = train_model(base_model_file=args["model_file"],
+                                    num_classes=len(class_mapping),
+                                    train_map_file=args["train_map_file"],
+                                    batch_size=args["batch_size"],
+                                    distributed_after=distributed_after,
+                                    num_epochs=args["epochs"],
+                                    freeze=True,
+                                    VERBOSE=VERBOSE)
+
+        if distributed.Communicator.rank() == 0:
+            if VERBOSE and not args["disable_timing"]:
+                print("Training time: {} seconds.".format(time.clock() - start))
+
+            trained_model.save(new_model_file)
+
+            if VERBOSE:
+                print("Stored trained model at '{}'.".format(new_model_file))
+
+    # Evaluation phase. We don't need to use multiple GPUs to evaluate, so we
+    # make only the first worker evaluate.
+    if distributed.Communicator.rank() == 0:
+
+        if args["evaluate_only"] and args["model_file"] == tl_model_file:
+            print("\n ERROR: Please provide a trained model to be used to "
+                  "evaluate, we cannot use the default model: '{}'.".format(
+                        tl_model_file
+                    ),
+                  file=sys.stderr)
+            exit(1)
+        elif args["evaluate_only"]:
```

Comment:
Consider to change this to a nested "if args["model_file"] == tl_model_file: ... else: ..."